### ISSUE TITLE
[question] Protobuf versioning scheme and Protobuf_VERSION variables

### ISSUE THREAD

**Issue by lia-viam** [conan-io/conan-center-index#25297]
### What is your question?

So the versioning schemes in protobuf are a tad confusing, see eg
https://protobuf.dev/support/version-support/#cpp
In the second table, a version like C++ version 5.26.x corresponds to protoc version 26.x.

But in a homebrew install, for example, `brew info protobuf` reports version `26.1`, and using `find_package(Protobuf CONFIG)` with such an install sets `Protobuf_VERSION` to `26.1.0`. But then `include/google/protobuf/port_def.inc` defines 
```cpp
#define PROTOBUF_VERSION 5026001
```
corresponding to `5.26.1`.

Now compare this to a `conan` project which `requires` and `tool_requires`, e.g., `protobuf/5.27.0`. The generated `FindProtobuf.cmake` sets `set(Protobuf_VERSION "5.27.0")`, but a `conan create .` with a CMake project which requires `protobuf/5.27.0` prints
```
...
-- Conan: Including build module from '/Users/lia/.conan2/p/b/proto24456bb2a2261/p/lib/cmake/protobuf/protobuf-module.cmake'
```
and in this file `protobuf-module.cmake`, we have `set(Protobuf_VERSION "27.0.0")`, which is also what's reported from
```shell
$ ~/.conan2/p/b/proto24456bb2a2261/p/bin/protoc --version
libprotoc 27.0
```

Anyway long story short, is there a canonical way to extract what we might call the `[lib]protoc` version from the conan installation? I do think conan may be doing the more correct thing here, but I am dealing with some existing CMake code which expected to get a 20-something variable number like the one produced by homebrew. Some options I was considering would be
- invoke `protoc --version` within CMake
- construct a new version number from `<minor>.<patch>.0` from the `Protobuf_VERSION`
- somehow include the `protobuf-module.cmake` ? 

Any feedback/suggestions are appreciated!

**Comment by Ahajha** [comment]
Potential option is to add it as a `conf_info` define (see https://docs.conan.io/2/reference/conanfile/methods/package_info.html#conf-info). Then you can just read this value from the consumer conanfile.

**Comment by AbrilRBS** [comment]
Hi @lia-viam thanks a lot for your question, sorry that this got no answer till now. 
A current workaround that can be implemented in the consumer conanfile, is to set the `system_package_version` property in your cmakedeps for protobuf, something like:
```
def generate(self):
    ....
    cmakedeps = CMakeDeps(self)
    cmakedeps.set_property("protobuf", "system_package_version", "20..your.desired.bit")
    cmakedeps.generate()
```

This will modify the config version generated for protobuf as part of your build (Note that we have https://github.com/conan-io/conan/pull/17504 that would add `cmake_config_version` as a property for this with a better naming)

Let me know if this can help, happy to help where possible :)
